system/src/system_network_internal.h

```diff
@@ -279,8 +279,10 @@ class ManagedNetworkInterface : public NetworkInterface
 
     void connect(bool listen_enabled=true) override
     {
-        INFO("ready():%s,connecting():%s,listening():%s",(ready())?"true":"false",(connecting())?"true":"false",(listening())?"true":"false");
-        if (!ready() && !connecting() && !listening())
+        // Do not try to connect if listening mode is active or pending
+        const bool listening = WLAN_SMART_CONFIG_START;
+        INFO("ready():%s,connecting():%s,listening():%s",(ready())?"true":"false",(connecting())?"true":"false",(listening)?"true":"false");
+        if (!ready() && !connecting() && !listening)
```

Comment:
This change allows to skip a connection attempt not only when listening mode is active, but also when it's pending. Otherwise, a connection initiated from the listening loop (https://github.com/spark/firmware/blob/feature/listen_mode_delay_fix/system/src/system_network_internal.h#L212) cannot be cancelled quickly via `network_listen()` call, due to the event loop entering `manage_network_connection()` before `manage_smart_config()` on its next iteration and blocking the system thread for another connection attempt.